site/blog/2022-10-28-optimize-s3-endpoints.mdx

```diff
@@ -0,0 +1,126 @@
+    <button class="button button--lg button--primary">Check your endpoints</button>
+  </a>
+</p>
+
+This query will find your active S3 buckets for all regions, associated with the existing endpoint gateways or interfaces - if they exist.
+
+Have you found missing endpoints? No problem, IaSQL can generate them for you. You can create two different
+types of endpoints: gateway and interface. We're going to describe their main features and you could create
+the desired type in an automated way.
+
+## Interface endpoints
+
+An interface endpoint is an elastic network interface with a private IP address that serves as an entry
+point for traffic destined to a supported AWS service. As they use ENIs, they can benefit of security groups
+to control traffic. Policies can also be added to control access to those endpoints.
+
+They are regional, meaning they can only be accessed by the same region they are created. Multi-region is
+possible using VPC peering, so resources in one region can be accessed from others. They only support IPv4
+TCP traffic.
+
+An interface endpoint (except S3 interface endpoint) has corresponding private DNS hostnames, that can be
+used to access the resources.
+
+Interface endpoints cover a wide range of services such as S3, Lambda, API Gateway, etc...
+
+<p>
+  <a href="https://app.iasql.com/#/button/Add S3 interface endpoints/--%20Inserts%20the%20missing%20endpoints%0Ainsert%20into%20endpoint_interface%28region%2C%20vpc_id%2C%20service%29%20select%20bucket.region%2C%20vpc.id%2C%20%27s3%27%20from%20bucket%20inner%20join%20vpc%0Aon%20bucket.region%3Dvpc.region%20where%20not%20exists%20%28select%20id%20from%20endpoint_interface%20where%20endpoint_interface.region%3Dbucket.region%20and%20endpoint_interface.vpc_id%3Dvpc.id%29%3B%0A--%20Preview%20the%20changes%0Aselect%20%2A%20from%20iasql_preview_apply%28%29%3B%0A--%20Apply%20the%20changes%0A--select%20%2A%20from%20iasql_apply%28%29%3B">
+    <button class="button button--lg button--primary">Add missing interface endpoint</button>
+  </a>
+</p>
+
+By clicking on the button, our engine will auto-generate the missing interface endpoints and will allow you
+to preview the changes to be applied on your cloud. <a href="https://docs.aws.amazon.com/vpc/latest/privatelink/create-interface-endpoint.html">Get more detail about interface endpoints.</a>
```

Comment:
```suggestion
By clicking on the button, our engine will auto-generate the missing interface endpoints and will
preview the changes to be applied on your cloud. Simply uncomment the final statement and run it to make the changes to your cloud account. <a href="https://docs.aws.amazon.com/vpc/latest/privatelink/create-interface-endpoint.html">Get more detail about interface endpoints.</a>
```

Reply:
Also, that last sentence with the link to interface endpoint documentation really doesn't flow in the post.